src/image.rs

```diff
@@ -1425,7 +1425,97 @@ impl Docker {
                         .header(CONTENT_TYPE, "application/json")
                         .header("X-Registry-Config", base64_url_encode(&ser_cred)),
                     Some(options),
-                    Ok(Full::new(root_fs)),
+                    Ok(body_full(root_fs)),
+                );
+                self.process_into_stream(req).boxed()
+            }
+            Err(e) => stream::once(async move { Err(e.into()) }).boxed(),
+        }
+        .map(|res| {
+            if let Ok(BuildInfo {
+                error: Some(error), ..
+            }) = res
+            {
+                Err(Error::DockerStreamError { error })
+            } else {
+                res
+            }
+        })
+    }
```

Comment:
Yeah I think we should combine these functions... I wonder if it's better to have `root_fs` as a `BodyType` or if that doesn't work, just go with the streaming version.

Reply:
So to be clear, this change is going to be a part of a semver-incompat version bump then? Seems unnecessary for a trivial change, but your call. 

I feel BodyType is best left as an implementation detail and not part of the public API--I think it's easier to consume an API that just takes the stream of bytes instead of having to figure out exactly how to construct this thing.

Reply:
Sure, feel free to keep it as you've done it. I can see your point here... We have already had one breaking change recently in the update container endpoint, so the next version needs to be a breaking one.